### ISSUE TITLE
Unexpected output after sampling using Conditional LDM

### ISSUE THREAD

**Issue by randomaccess2023** [explainingai-code/StableDiffusion-PyTorch#9]
I am getting a bunch of noodle-like waves after sampling for conditional LDM instead of proper digits. The unconditional LDM works fine. I am using the MNIST dataset that the Torchvision library has ([torchvision.datasets.MNIST](https://pytorch.org/vision/main/generated/torchvision.datasets.MNIST.html#torchvision.datasets.MNIST)).

**Can you tell what could be wrong in this scenario?**

I have attached `x0_0.png` outputs for both __Unconditional_LDM__ and __Conditional_LDM__.
![Unconditional_LDM_x0_0](https://github.com/explainingai-code/StableDiffusion-PyTorch/assets/84232498/927727cd-93c0-46d5-a5e0-ce807b2f6a5c)
![Conditional_LDM_x0_0](https://github.com/explainingai-code/StableDiffusion-PyTorch/assets/84232498/41a8284d-0599-4f78-821f-2697546ee27a)

**Comment by explainingai-code** [comment]
Hi @randomaccess2023 , 
Can you share the config that you have and also the sampling script (if you have made any changes in that) ? and also shape of xt you get [here](https://github.com/explainingai-code/StableDiffusion-PyTorch/blob/main/tools/sample_ddpm_class_cond.py#L26) 
This seems like an issue of the model being asked to generate images of different size that it was trained with. So just what to confirm if that is somehow not the case.

**Comment by randomaccess2023** [comment]
> Also I think the model is getting images scaled from 0-1 rather than -1 to 1. If thats the case then after loading the mnist images make sure to do the scaling as well(similar to https://github.com/explainingai-code/StableDiffusion-PyTorch/blob/main/dataset/mnist_dataset.py#L91 )

@explainingai-code Yes, I scaled the images within a range of 0 to 1 rather than -1 to 1. I always scale from 0 to 1. I changed it and it resulted in a **higher codebook loss for VQ-VAE** after training for 10 epochs. Now, I am training the LDM which will take more time for 100 epochs. Also, I fed the labels directly to the embedding table and then added them to the position encoding layer. It worked for Conditional DDPM and for that reason, I didn't use the one-hot method.

**Comment by explainingai-code** [comment]
Yeah that way of conditioning is also fine.
